bundler/lib/bundler/source/rubygems.rb

```diff
@@ -314,10 +314,11 @@ def credless_remotes
       end
 
       def remotes_for_spec(spec)
-        specs.search_all(spec.name).inject([]) do |uris, s|
+        uris = []
+        specs.with_all_matching(spec.name) do |s|
```

Comment:
this was the only caller of search_all, and in just injected the result to make a list of arrays. It's way less overhead if we directly yield each search result.

Reply:
I figured out a way to preserve most of the old code while still avoiding extra concats. Reverted this change.